catalyst-gateway/bin/src/settings/str_env_var.rs

```diff
@@ -0,0 +1,272 @@
+//! Processing for String Environment Variables
+use std::{
+    env::{self, VarError},
+    fmt::{self, Display},
+    str::FromStr,
+};
+
+use strum::VariantNames;
+use tracing::{error, info};
+
+/// An environment variable read as a string.
+#[derive(Clone)]
+pub(crate) struct StringEnvVar {
+    /// Value of the env var.
+    value: String,
+    /// Whether the env var is displayed redacted or not.
+    redacted: bool,
+}
+
+/// Ergonomic way of specifying if a env var needs to be redacted or not.
+pub(super) enum StringEnvVarParams {
+    /// The env var is plain and should not be redacted.
+    Plain(String, Option<String>),
+    /// The env var is redacted and should be redacted.
+    Redacted(String, Option<String>),
+}
+
+impl From<&str> for StringEnvVarParams {
+    fn from(s: &str) -> Self {
+        StringEnvVarParams::Plain(String::from(s), None)
+    }
+}
+
+impl From<String> for StringEnvVarParams {
+    fn from(s: String) -> Self {
+        StringEnvVarParams::Plain(s, None)
+    }
+}
+
+impl From<(&str, bool)> for StringEnvVarParams {
+    fn from((s, r): (&str, bool)) -> Self {
+        if r {
+            StringEnvVarParams::Redacted(String::from(s), None)
+        } else {
+            StringEnvVarParams::Plain(String::from(s), None)
+        }
+    }
+}
+
+impl From<(&str, bool, &str)> for StringEnvVarParams {
+    fn from((s, r, c): (&str, bool, &str)) -> Self {
+        if r {
+            StringEnvVarParams::Redacted(String::from(s), Some(String::from(c)))
+        } else {
+            StringEnvVarParams::Plain(String::from(s), Some(String::from(c)))
+        }
+    }
+}
+
+/// An environment variable read as a string.
+impl StringEnvVar {
```

Comment:
I think we need some unit tests for this type

Reply:
I agree, but I am not writing them.